[8.0] port of module logistic_requisition

yvaucher opened the pull request:
Here is a first PR porting the code of logistic_requisition to 8.0 api and to new 8.0 workflow

Some tests are commented as they need to be adapted to new 8.0 workflow.

It also:
- breaks link between logistic requisition source and sale order line to simplify the code and to add flexibility
- ports webkit report to Qweb
- removes dependancy on transport_plan

depends on
- [x] https://github.com/OCA/sale-workflow/pull/49
- [x] https://github.com/OCA/purchase-workflow/pull/26
- [x] https://github.com/OCA/purchase-workflow/pull/36

yvaucher committed:
reactivate logistic_requisition module

yvaucher committed:
reformat and add missing license headers

yvaucher committed:
remove transport_plan dependancy

yvaucher committed:
port models and method to API 8.0

yvaucher committed:
refactor workflows

yvaucher committed:
remove eta things as transport_plan is not used

yvaucher committed:
define default values

yvaucher committed:
port onchanges

yvaucher committed:
move onchange_validate on logistic_requisition_budget

yvaucher committed:
port xmlid_to_object

yvaucher committed:
add dependancy on stock_dropshipping

yvaucher committed:
improve manifest

yvaucher committed:
Remove additions on purchase order as product service do not generate procurement anymore and drop shipping automatically validates it's procurement

yvaucher committed:
break link with LR

set cost estimate lines to manually sourced and ensure user has set a po line

yvaucher committed:
type of sol in sale order form does not exist anymore

yvaucher committed:
remove stock_owner

yvaucher committed:
remove board_purchase_form which was removed on odoo/c60c1d1a725

yvaucher committed:
fix menuitem to not have twice base menu Order Management

yvaucher committed:
dependancy purchase_requisition_extended renamed to purchase_requisition_bid_selection

yvaucher committed:
restore tests for migration to 8.0

yvaucher committed:
remove dead code _create_pickings_and_procurements as stock_drop_shipping and need procurement groups are doing the job of spliting sale order into multiple procurement

yvaucher committed:
remove overide of copy to avoid the copy of invoice_ids as it is implemented in base sale.py

yvaucher committed:
adapt tests to breaking link between SOL and LRS

yvaucher committed:
add dependancy on sale_sourced_by_line

yvaucher committed:
remove transient model access rules

yvaucher committed:
restrict selection of po line on cost estimate line to draft and confirm

yvaucher committed:
extend is_source to done purchase requisitions

yvaucher committed:
port mako report to Qweb

gurneyalex commented:
could you rename ./logistic_requisition/tests/test_mutlicurrency_update_po_line.py to ./logistic_requisition/tests/test_multicurrency_update_po_line.py?

yvaucher commented:
ok mu **lt** i

yvaucher committed:
fix mu(tl)iple typo apearing 3 times

yvaucher committed:
remove old report template

yvaucher committed:
[PEP8] in tests

lepistone committed:
lint

lepistone committed:
fix mutable default parameter

lepistone committed:
remove deprecated fast_suite and checks

lepistone committed:
bump version: module changed substantially in v8

lepistone committed:
travis: add dependency on purchase-workflow

lepistone committed:
lint

jgrandguillaume commented:
Hi,

The button "button_create_cost_estimate" from the LR form is not working properly. 

a) Create a LR with 2 lines, confirm LR,  source lines with whatever you want (dispatch is the easiest way)
b) Mark all LR line as "sourced"
c) Try to create a coste estimate using the button on LR form => not working. BUT is I open the line using "view lines" button on LR form, select them all, chose "More->create cost estimate" it's working.

Regards,

Joël

gurneyalex commented:
activity_code, account_code: this is something customer specific, which should not be in this project -> to remove

yvaucher commented on logistic_requisition/model/logistic_requisition.py:
```diff
@@ -0,0 +1,1067 @@
+        readonly=True)
+    note = fields.Text('Notes')
+    activity_code = fields.Char('Activity Code', size=32)
+    account_code = fields.Char('Account Code', size=32)
```
activity_code and account_code to remove as it is too specific

yvaucher commented on logistic_requisition/test/logistic_requisition_report_test.yml:
```diff
@@ -10,8 +10,10 @@
+        requested_uom_id: product.product_uom_unit
         date_delivery: !eval "time.strftime('%Y-%m-%d')"
         logistic_user_id: base.user_demo
         account_code: 'Shazam!'
```
to remove from test as well

yvaucher commented on logistic_requisition/tests/test_mto_workflow.py:
```diff
@@ -65,41 +63,48 @@ class test_mto_workflow(common.TransactionCase):
-            'requested_uom_id': self.product_uom_pce,
+            'requested_uom_id': self.product_uom_pce.id,
             'date_delivery': time.strftime(D_FMT),
             'account_code': 'a code',
```
to remove from test as well

yvaucher commented on logistic_requisition/tests/test_mto_workflow.py:
```diff
@@ -183,73 +199,67 @@ def _check_sale_line(self, sale_line):
-            'requested_uom_id': self.product_uom_pce,
+            'requested_uom_id': self.product_uom_pce.id,
             'date_delivery': time.strftime(D_FMT),
             'account_code': 'a code',
```
to remove from test as well

yvaucher committed:
remove account_code and activity_code as non generic

lepistone committed:
travis: add server-tools for web_context_tunnel

yvaucher committed:
fix create cost estimate buttons

yvaucher commented:
@jgrandguillaume button fixed

yvaucher commented:
@gurneyalex extra fields moved

lepistone committed:
requisition date -> datetime

This is to be consistent with a last-minute change in odoo 8 that made
the date of a purchase order a datetime field.

yvaucher committed:
break relation on country_id to country of consignee
and add an onchange on consignee to set it

yvaucher committed:
add total amount in company currency on requisition line

yvaucher committed:
move description in README.rst

lepistone committed:
add fields: transport mode and transport vehicle

lepistone committed:
add dependency for transport documents

yvaucher committed:
fix _do_done method to accept multiple records

gurneyalex committed:
fix LR sourcing with a PR

* use the correct API version in the wizard:
  the code would use self.pool.get which returns an old-API model
  and then use it through the new API

* add a yaml test for this sourcing flow (long because we need to manually
  process the whole PR)

* fix various isssues with new API use:
  - cannot write on a recorset with > 1 records
  - confusions with bits expecting records / ids

gurneyalex commented:
sourcing from PR is completely broken. I'm on it (and adding the missing test)

yvaucher commented:
@gurneyalex warning it seams you added few temp files in last commit

lepistone committed:
fix regression because of a comma too many

gurneyalex committed:
don't change purchase_requisition view

if we hide these buttons, we cannot complete the sourcing process

gurneyalex committed:
removed dead code in the test

gurneyalex committed:
test cleanup

gurneyalex commented:
:+1: I spent lots of time reviewing this and testing, and I think most of the ground is covered. What remains can certainly be processed using bug fixing PRs. Waiting  for Travis to flag my last commit as green and I'll merge.

yvaucher commented:
agreed :+1:

yvaucher commented on logistic_requisition/model/requisition_cancel.py~HEAD:
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
```
some cleaning to do this file shouldn't be there
..stic_requisition/model/requisition_cancel.py~HEAD

guewen commented:
![Applause!](http://gifs.joelglovier.com/clap/applause.gif)

gurneyalex commented:
:lollipop:

lepistone commented:
Thanks for taking care of that! I :+1: the merge FWIW.

yvaucher commented:
:tada:
